engine/src/communication/messages/GPUComponentMessage.h

```diff
@@ -189,71 +157,78 @@ class GPUComponentMessage : public GPUMessage {
+				char * charsPointer = (char *) raw_buffers[columns_offsets[i].strings_data];
+				cudf::size_type * offsetsPointer = (cudf::size_type *) raw_buffers[columns_offsets[i].strings_offsets];
+				cudf::bitmask_type * nullMaskPointer = nullptr;
 				if(columns_offsets[i].strings_nullmask != -1) {
-					nullMaskPointer = (unsigned char *) raw_buffers[columns_offsets[i].strings_nullmask];
+					nullMaskPointer = (cudf::bitmask_type *) raw_buffers[columns_offsets[i].strings_nullmask];
+				} 
+				// auto unique_column = std::make_unique<cudf::column>(dtype, column_size, data_buff, valid_buff);
+				cudf::size_type column_size  =  (cudf::size_type)columns_offsets[i].metadata.size;
+				
+				rmm::device_vector<char> d_strings(charsPointer, charsPointer + columns_offsets[i].strings_data_size);
+				rmm::device_vector<cudf::size_type> d_offsets(offsetsPointer, offsetsPointer + columns_offsets[i].strings_offsets_size/sizeof(cudf::size_type));
+				rmm::device_vector<cudf::bitmask_type> d_null_mask{};
+				if (nullMaskPointer != nullptr) {
+				 	d_null_mask = rmm::device_vector<cudf::bitmask_type>(nullMaskPointer, nullMaskPointer + cudf::bitmask_allocation_size_bytes(column_size)/sizeof(cudf::bitmask_type));
 				}
-
-				auto nvcategory_ptr = NVCategory::create_from_offsets(reinterpret_cast<const char *>(stringsPointer),
-					columns_offsets[i].metadata.size,
-					reinterpret_cast<const int *>(offsetsPointer),
-					reinterpret_cast<const unsigned char *>(nullMaskPointer),
-					columns_offsets[i].metadata.null_count,
-					true);
-				received_samples[i].create_gdf_column(
-					nvcategory_ptr, columns_offsets[i].metadata.size, (char *) columns_offsets[i].metadata.col_name);
-
-				RMM_TRY(RMM_FREE(stringsPointer, 0));
-				RMM_TRY(RMM_FREE(offsetsPointer, 0));
-				RMM_TRY(RMM_FREE(nullMaskPointer, 0));
+				cudf::size_type null_count = columns_offsets[i].metadata.null_count;
+				auto unique_column = cudf::make_strings_column(d_strings, d_offsets, d_null_mask, null_count);
+				received_samples[i] = std::move(unique_column);
 			} else {
+				cudf::data_type dtype = cudf::data_type{cudf::type_id(columns_offsets[i].metadata.dtype)};
+				cudf::size_type column_size  =  (cudf::size_type)columns_offsets[i].metadata.size;
+
 				cudf::valid_type * valid_ptr = nullptr;
+				cudf::size_type valid_size = 0;
 				if(columns_offsets[i].valid != -1) {
 					// this is a valid
 					auto valid_offset = columns_offsets[i].valid;
 					valid_ptr = (cudf::valid_type *) raw_buffers[valid_offset];
+					valid_size = cudf::bitmask_allocation_size_bytes(column_size);
+				}
+				assert(raw_buffers[data_offset] != nullptr);
+				try {
+					auto dtype_size = cudf::size_of(dtype);
+					auto buffer_size = column_size * dtype_size;
+					rmm::device_buffer data_buff(raw_buffers[data_offset], buffer_size);
+					rmm::device_buffer valid_buff(valid_ptr, valid_size);
+					auto unique_column = std::make_unique<cudf::column>(dtype, column_size, data_buff, valid_buff);
```

Comment:
Need to remember to use std::move for all data holding/owning data structures. For ALL device_buffers we use and create we want to move them around. Maybe even put them in a unique_ptr when you create them and then release them when finally moving them into a column

Reply:
ok!